web/tests/test_tournament.py

```diff
@@ -4,52 +4,296 @@
 
 from flask import url_for
```

Comment:
nit for this file: in a future commit, this should probably live as `web/tournament/test_tournament.py`. But we haven't been structuring are tests this way, so let's save this for another PR